staging/src/k8s.io/apiserver/pkg/authentication/authenticator/interfaces.go

```diff
@@ -77,4 +78,37 @@ type Response struct {
 	Audiences Audiences
 	// User is the UserInfo associated with the authentication context.
 	User user.Info
+	// AuthenticatorName identifies the authenticator that authenticated the request.
+	AuthenticatorName string
```

Comment:
Does the overall response ever get logged, i.e. in the case of audit?

Reply:
Could you please explain why this matters in the context of this PR.

Reply:
the expected use and meaning of this should be documented... for example:
* is this user-facing?
* is this included in apiserver logs or audit logs?
* is this included in metrics?
* is this expected to be human-readable? stable over time?
* is this passed to the authorization layer?
* is this required? does it affect authentication success or failure?

Reply:
this also seems to be mixing two dimensions: authentication method (token, basic, x509, requestheader, anonymous, etc) and authenticator id (really only needed for token auth, e.g. oidc, legacy-serviceaccount, bound-serviceaccount, webhook, token-file, apiserver-loopback, etc)

Reply:
@liggitt 
I updated the description of this PR with the bullet points that your provided.
I left a couple of them with "?" marks:
1. is this included in apiserver logs or audit logs?
2. is this passed to the authorization layer?
I think it is a "no" for both, but I am 100% sure.

On the intuitive level, I believe that I understand what you mean by "mixing two dimensions". However, I am not clear on what that means in terms of labels.
Are you suggesting that the Response type should have two fields to properly identify the authenticator that produced it: AuthMethod and AuthID?
As an example, Responses produced by the token-file authenticator should be labeled as 
- AuthMethod: Token
- AuthID: File

And Responses produced by the x509 authenticator should labeled as AuthMethod: x509 (no AuthID).

Reply:
> I updated the description of this PR with the bullet points that your provided

it might be good to include those aspects in the godoc as well

> * is this included in apiserver logs or audit logs?
> * is this passed to the authorization layer?
>     I think it is a "no" for both, but I am 100% sure.

I wasn't sure if we wanted that included in audit logs as an audit annotation or not.

> On the intuitive level, I believe that I understand what you mean by "mixing two dimensions". However, I am not clear on what that means in terms of labels.
> Are you suggesting that the Response type should have two fields to properly identify the authenticator that produced it: AuthMethod and AuthID?
> As an example, Responses produced by the token-file authenticator should be labeled as
> 
> * AuthMethod: Token
> * AuthID: File

The specific names we pick for the different methods and different authenticators are TBD, but something along those lines, yes.

> And Responses produced by the x509 authenticator should labeled as AuthMethod: x509 (no AuthID).

I would actually go ahead and pick a name for the built-in client-cert authenticator (e.g. `client-ca-file`) in case we ever add a second type.

Reply:
Add AuthMethod field to both Response and Error.
PTAL.

Reply:
Updated godoc, PTAL.